test/functional/test_framework/test_framework.py

```diff
@@ -489,17 +489,19 @@ def set_dash_test_params(self, num_nodes, masterodes_count, extra_args, fast_dip
         self.setup_clean_chain = True
         self.is_network_split = False
         # additional args
+        if extra_args is None:
+            extra_args = [[] * num_nodes]
+        assert_equal(len(extra_args), num_nodes)
         self.extra_args = extra_args
-
-        self.extra_args += ["-sporkkey=cP4EKFyJsHT39LDqgdcB43Y3YXjNyjb5Fuas1GQSeAtjnZWmZEQK"]
-
+        self.extra_args[0] += ["-sporkkey=cP4EKFyJsHT39LDqgdcB43Y3YXjNyjb5Fuas1GQSeAtjnZWmZEQK"]
```

Comment:
Was this previously setting the spork key on all nodes? (E.g. applying `extra_args` to all?)

Reply:
Yep, technically this is a change in behaviour 🙈 But `spork` rpc is never used to set sporks from nodes other than the control one (in `DashTestFramework`, `spork` and `multispork` tests use `BitcoinTestFramework` deal with this manually).